### PR TITLE
[Domain Purchase] Update domain empty view UI

### LINKED ISSUE #20075: Update domain empty view UI
The empty view of the domain screen in site creation flow will be updated.
The view is displayed after the `Choose a theme` screen, when the search input is empty.

#### Subtasks

- [ ] Update title and subtitle text and left align it
- [ ] Update search input
- [ ] Change illustration view

### Design

| Before | After |
| - | - |
|  |  |

---
> **Note**:
> - Other changes are out of scope and part of other tasks
> - All development should be behind [the feature flag](https://github.com/wordpress-mobile/WordPress-Android/issues/17896)

### PR DESCRIPTION
**Pull request by alpavanoglu**
Fixes #20075

This PR updates the empty view UI. The navigation button colors will be handled separately along with dark mode colors as they are not there yet. There might be mismatch between the colors and the designs due to missing information. We should focus on colors along with dark mode changes.

### Flag Enabled
1. Install & Launch & Login Jetpack
2. Enable `Site Creation Domain Purchase` flag
3. From **Home** Tap on the downward arrow icon next to the site-name to open the menu
4. Tap on ➕  icon from top right -> **Create WordPress.com site**
5. Navigate until "Choose a domain" screen
6. Verify if the UI matches as per design.
7. Interact with the search bar and proceed to the next screen. Everything should be connected properly.

### Flag Disabled
1. Follow the steps from above until (6).
2. Make sure the screen is exactly like the original, unaffected.

## Regression Notes
1. Potential unintended areas of impact
N/A

8. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

9. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by salimbraksa** [comment]
Thanks for your work on this @alpavanoglu. The new empty view looks pretty good but it would be even better if we can polish it a bit more, long as it's not too difficult to do in SwiftUI.

⚠️ Issues that require your attention.
🔍 Issues that are just a nitpick are you are free to ignore them.

| Screenshot | Description |
| ------------ | ----------- |
|  | ⚠️ The border is entirely missing in the left and right corners of the view. And the top-border looks slightly thicker compared to other edges. |
|  | ⚠️ The view doesn't render well in Dark Mode. |
| <video src="https://user-images.githubusercontent.com/9609223/225963096-e320c3f5-8803-4d71-81e0-4ea9ca0507ca.mp4" ></video> | ⚠️ The app crashes when rotating the device from portrait to landscape and vice-versa. I'm not entirely sure it's related to this PR, but when I disabled the `Domain Purchasing` flag, the crash is gone. |
| <video src="https://user-images.githubusercontent.com/9609223/225964010-8c846690-0211-456b-a678-14a77e37039a.mp4" ></video> | 🔍 The empty view takes a second to appear. I think it should appear immediately when the "Choose a domain" screen appears. I haven't seen the code changes but it looks like the empty view appears on `viewDidAppear` and not on `viewDidLoad`. |

**Comment by mokagio** [comment]
I'll move the milestone to 22.1 because I'm about to start the 22.0 code freeze and this hasn't been reviewed yet. Feel free to revert if necessary.

File: WordPress/Classes/ViewRelated/Site Creation/Web Address/DesignSystem/Color+DesignSystem.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        enum Foreground {
+            static let primary = Color("foregroundPrimary")
+            static let secondary = Color("foregroundSecondary")
+            static let tertiary = Color("foregroundTertiary")
+        }
+
+        enum Background {
+            static let primary = Color("backgroundPrimary")
+            static let secondary = Color("backgroundSecondary")
+            static let tertiary = Color("backgroundTertiary")
```
Comment (alpavanoglu) [inline WordPress/Classes/ViewRelated/Site Creation/Web Address/DesignSystem/Color+DesignSystem.swift:14]: Before we use these outside of this feature, we need to align with the designers. After we can values from them, I think we can start using these pretty safely. The `DS` namespace will guard us against mixing these up with the old values. We should probably also add the same constants to `UIColor`

File: WordPress/Classes/ViewRelated/Site Creation/Web Address/SiteCreationEmptySiteTemplate.swift [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    enum Strings {
+        static let tooltipSiteName = NSLocalizedString(
+            "YourSiteName.com",
+            comment: "Site name that is placed in the tooltip view."
+        )
+
+        static let tooltipDescription = NSLocalizedString(
+            "Like the example above, a domain allows people to find and visit your site from their web browser.",
+            comment: "Site name description that sits in the template website view."
+        )
```
Comment (salimbraksa) [inline WordPress/Classes/ViewRelated/Site Creation/Web Address/SiteCreationEmptySiteTemplate.swift:127]: Just a reminder that it's recommended to use reverse-dns for string localizations. At least for brand new strings. p1657627987789469-slack-C011BKNU1V5

File: WordPress/Classes/ViewRelated/Site Creation/Web Address/WebAddressWizardContent.swift [hunk 3]
```diff
@@ -35,7 +36,10 @@ final class WebAddressWizardContent: CollapsableHeaderViewController {
     private let table: UITableView
     private let searchHeader: UIView
     private let searchTextField: SearchTextField
+    private let searchBar = UISearchBar()
     private var sitePromptView: SitePromptView!
+    private let siteCreationEmptyTemplate = SiteCreationEmptySiteTemplate()
+    private lazy var hostingController = UIHostingController(rootView: siteCreationEmptyTemplate)
```
Comment (salimbraksa) [inline WordPress/Classes/ViewRelated/Site Creation/Web Address/WebAddressWizardContent.swift]: nit: `hostingController` naming is a bit generic, would be great to rename it to something related to `siteCreationEmptyTemplate` view. So the reader knows that the `hostingController` is for the `Empty Site` view.

**Review by salimbraksa — APPROVED** [review]

**Pull request merged**